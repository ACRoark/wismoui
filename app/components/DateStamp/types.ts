export type DateStampFormat = 'auto' | 'long' | 'short';
