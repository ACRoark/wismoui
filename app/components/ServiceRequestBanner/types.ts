export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
