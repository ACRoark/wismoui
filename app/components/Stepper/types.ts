import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
