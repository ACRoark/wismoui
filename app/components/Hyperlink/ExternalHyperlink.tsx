/**
 * A link to a certain page, an anchor tag
 */
import React from 'react';

import ExternalLinkIcon from 'images/ExternalLinkIcon';
import { IBaseHyperlinkProps } from './types';

const ExternalHyperlink: React.FC<IBaseHyperlinkProps> = (props: IBaseHyperlinkProps): React.ReactElement => {
  const { children, href, target } = props;

  return (
    <a href={`${href}`} target={target || '_blank'}>
      <ExternalLinkIcon />
      {React.Children.toArray(children)}
    </a>
  );
};

export default ExternalHyperlink;
