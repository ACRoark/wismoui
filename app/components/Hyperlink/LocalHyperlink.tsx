/**
 * A link to a certain page, an anchor tag
 */
import React from 'react';
import { Link } from 'react-router-dom';
import { IBaseHyperlinkProps } from './types';

const LocalHyperlink: React.FC<IBaseHyperlinkProps> = (props: IBaseHyperlinkProps): React.ReactElement => {
  const { children, href, target } = props;

  return (
    <Link to={`${href}`} target={target || '_self'}>
      {React.Children.toArray(children)}
    </Link>
  );
};

export default LocalHyperlink;
