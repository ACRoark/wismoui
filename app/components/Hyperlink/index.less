@import (reference) '../../styles/themes/index.less';

@hyperlink-cls-prefix: ~'@{dte-prefix}-hyperlink';

.@{hyperlink-cls-prefix}-icon {
  line-height: 16px;
  margin-right: 10px;
  width: 16px;
}
