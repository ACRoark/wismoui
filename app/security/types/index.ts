import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
