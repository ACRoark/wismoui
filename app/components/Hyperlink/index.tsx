/**
 * A link to a certain page, an anchor tag
 */
import React from 'react';

import './index.less';

import ExternalHyperlink from './ExternalHyperlink';
import LocalHyperlink from './LocalHyperlink';
import { IBaseHyperlinkProps } from './types';

const isLocalUrl = (href: string): boolean => {
  const regex = /(http|https):\/\/.*/;

  return !regex.test(href);
};

const Hyperlink: React.FC<IBaseHyperlinkProps> = (props: IBaseHyperlinkProps): React.ReactElement => {
  const { children, href, target } = props;

  const isLocal = isLocalUrl(href);

  if (isLocal) {
    return (
      <LocalHyperlink href={`${href}`} target={target || '_self'}>
        {children}
      </LocalHyperlink>
    );
  }

  return (
    <ExternalHyperlink href={`${href}`} target={target || '_blank'}>
      {children}
    </ExternalHyperlink>
  );
};

export default Hyperlink;
