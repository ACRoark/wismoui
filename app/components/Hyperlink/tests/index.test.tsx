import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { createSnapshot } from 'testing/utils';

import Hyperlink from '..';
import { BrowserTarget } from '../types';

const testHyperlink = (href: string, text: string, target?: BrowserTarget): void => {
  const tree = createSnapshot(
    <Router>
      <Hyperlink href={href} target={target}>{text}</Hyperlink>
    </Router>
  );

  expect(tree).toMatchSnapshot();
};

const testExternalHyperlink = (target?: BrowserTarget): void => testHyperlink('https://www.google.com', 'External Link Text', target);

const testLocalHyperlink = (target?: BrowserTarget): void => testHyperlink('/', 'Local Link Text', target);

describe('<Hyperlink />', (): void => {
  it('should render external link when href is for a different host', (): void => testExternalHyperlink());

  it('should render internal link when href is for the current host', (): void => testLocalHyperlink());

  describe('target', (): void => {
    it('should be set to _blank when a target is not provided and href is for a different host', (): void => testExternalHyperlink());

    it('should be set to _self when a target is not provided and href is for the current host', (): void => testLocalHyperlink());

    it('should be set to the given target when a target is provided and href is for a different host', (): void => testExternalHyperlink('_top'));

    it('should be set to the given target when a target is provided and href is for the current host', (): void => testLocalHyperlink('_top'));
  });
});
