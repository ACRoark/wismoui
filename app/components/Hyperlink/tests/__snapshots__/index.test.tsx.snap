// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hyperlink /> should render external link when href is for a different host 1`] = `
<a
  href="https://www.google.com"
  target="_blank"
>
  <svg
    className="dte-wismo-hyperlink-icon"
    fill="currentColor"
    height="16"
    role="img"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      id="external-link"
    >
      <path
        d="M5.75 3C11.1197 3 14 3 14 3M14 3V11.25M14 3L3 14"
        id="Vector"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </g>
  </svg>
  External Link Text
</a>
`;

exports[`<Hyperlink /> should render internal link when href is for the current host 1`] = `
<a
  href="/"
  onClick={[Function]}
  target="_self"
>
  Local Link Text
</a>
`;

exports[`<Hyperlink /> target should be set to _blank when a target is not provided and href is for a different host 1`] = `
<a
  href="https://www.google.com"
  target="_blank"
>
  <svg
    className="dte-wismo-hyperlink-icon"
    fill="currentColor"
    height="16"
    role="img"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      id="external-link"
    >
      <path
        d="M5.75 3C11.1197 3 14 3 14 3M14 3V11.25M14 3L3 14"
        id="Vector"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </g>
  </svg>
  External Link Text
</a>
`;

exports[`<Hyperlink /> target should be set to _self when a target is not provided and href is for the current host 1`] = `
<a
  href="/"
  onClick={[Function]}
  target="_self"
>
  Local Link Text
</a>
`;

exports[`<Hyperlink /> target should be set to the given target when a target is provided and href is for a different host 1`] = `
<a
  href="https://www.google.com"
  target="_top"
>
  <svg
    className="dte-wismo-hyperlink-icon"
    fill="currentColor"
    height="16"
    role="img"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      id="external-link"
    >
      <path
        d="M5.75 3C11.1197 3 14 3 14 3M14 3V11.25M14 3L3 14"
        id="Vector"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </g>
  </svg>
  External Link Text
</a>
`;

exports[`<Hyperlink /> target should be set to the given target when a target is provided and href is for the current host 1`] = `
<a
  href="/"
  onClick={[Function]}
  target="_top"
>
  Local Link Text
</a>
`;
